src/popup/components/TransactionFilter.vue

```diff
@@ -0,0 +1,142 @@
+<template>
+  <div
+    v-if="showFilterBar"
+    class="transaction-filter"
+    :class="{ 'without-search': hideSearch }"
+    :style="{ maxHeight: openHeight }"
+  >
+    <transition
+      name="fade-transition"
+      mode="out-in"
+    >
+      <div
+        v-if="firstRender || openHeight"
+        ref="transactionFilter"
+        class="filter-wrapper"
+      >
+        <InputSearch
+          v-if="!hideSearch"
+          v-model="searchPhrase"
+          class="input-search"
+          :placeholder="$t('pages.fungible-tokens.searchPlaceholder')"
+          @focusState="handleFocus"
+        />
+        <Filters
+          v-if="!hideFilterButton"
+          v-model="displayMode"
+          class="filters"
+          :filters="filtersConfig"
+          :scroll-top-threshold="scrollTopThreshold"
+        />
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script lang="ts">
+import {
+  computed,
+  defineComponent,
+  onMounted,
+  ref, watch,
+} from '@vue/composition-api';
+import {
+  useTransactionFiler,
+} from '../../composables';
+import InputSearch from './InputSearch.vue';
+import Filters from './Filters.vue';
+import { IFilters, ObjectValues } from '../../types';
+
+export default defineComponent({
+  name: 'TransactionFilter',
+  components: { Filters, InputSearch },
+  props: {
+    showFilters: Boolean,
+  },
+  setup(props, { root, refs }) {
+    const scrollTopThreshold = 140;
+    const maxHeight = ref(0);
+    const firstRender = ref(true);
+    const inputIsFocused = ref(false);
+
+    const {
+      expandSearchBarAndFilter,
+      searchPhrase,
+      displayMode,
+      FILTER_MODE,
+    } = useTransactionFiler();
+
+    const filtersConfig = ref<IFilters<ObjectValues<typeof FILTER_MODE>>>({
+      all: { name: root.$t('filters.all') },
+      in: { name: root.$t('filters.in') },
+      out: { name: root.$t('filters.out') },
+      dex: { name: root.$t('filters.dex') },
+    });
+
+    const openHeight = computed(() => props.showFilters
+      || inputIsFocused.value
+      || expandSearchBarAndFilter.value ? `${maxHeight.value}px` : 0);
```

Comment:
In case of multiline conditions I suggest to put every item in new line and wrap the conditions in a brackets:
```
=> (
  test1
  && test2
  && test3
)
  ? result1
  : result2
```